packages/gestalt/src/Modal.js

```diff
@@ -34,16 +37,29 @@ const SIZE_WIDTH_MAP = {
   lg: 900,
 };
 
-function Header({ heading }: {| heading: string | Node |}) {
+function Header({
+  align,
+  heading,
+  subHeading,
+}: {|
```

Comment:
can we try segmenting flow here

type Header = {| heading:  Node |} | {| align: 'left' | 'center',
  heading: string,
  subHeading?: string,|}

Reply:
hope this works too. it might require playing around a little.

Reply:
I got some flow errors doing this, but I was able to get it to work if I moved the string check to the component. Let me know if you think this is a better way!

Reply:
yeah makes sense!